lsp/src/main/scala/org/ensime/lsp/ensime/EnsimeLanguageServer.scala

```diff
@@ -255,10 +255,12 @@ class EnsimeLanguageServer(in: InputStream, out: OutputStream)
 
   override def shutdown(): Unit = {
     log.info("Shutdown request")
-    system.terminate()
-    log.info("Shutting down actor system.")
-    Await.result(system.whenTerminated, Duration.Inf)
-    log.info("Actor system down.")
+    if (!(system eq null)) {
+      system.terminate()
+      log.info("Shutting down actor system.")
+      Await.result(system.whenTerminated, Duration.Inf)
+      log.info("Actor system down.")
```

Comment:
feel free to stick a `sys.exit` in here

Reply:
actually, gerp the codebase for the other place where we do that. We have a system property guard so we don't exit the JVM when running tests.

Reply:
As in [Server.scala](https://github.com/ensime/ensime-server/blob/3.0/server/src/main/scala/org/ensime/server/Server.scala#L211-L215)?  The bifurcation of servers [here](https://github.com/ensime/ensime-server/blob/3.0/server/src/main/scala/org/ensime/server/Server.scala#L160-L162) guarantees the existence of the actor system in the "regular" but not the "lsp".  A similar null check can be found  [here](https://github.com/ensime/ensime-server/blob/3.0/lsp/src/main/scala/org/ensime/lsp/ensime/EnsimeLanguageServer.scala#L193) and [here](https://github.com/ensime/ensime-server/blob/3.0/lsp/src/main/scala/org/ensime/lsp/ensime/EnsimeLanguageServer.scala#L202)

Reply:
I mean a check to see if you're in test mode or not, and then just exit the JVM.

Reply:
I think `sys.exit` should be called in the `exit` notification handler, not the `shutdown` request. See https://github.com/ensime/ensime-server/pull/1946.